app/models/agents/trigger_agent.rb

```diff
@@ -75,26 +75,30 @@ def receive(incoming_events)
           rule_values = rule['value']
           rule_values = [rule_values] unless rule_values.is_a?(Array)
 
-          rule_values.any? do |rule_value|
-            case rule['type']
-            when "regex"
-              value_at_path.to_s =~ Regexp.new(rule_value, Regexp::IGNORECASE)
-            when "!regex"
-              value_at_path.to_s !~ Regexp.new(rule_value, Regexp::IGNORECASE)
-            when "field>value"
-              value_at_path.to_f > rule_value.to_f
-            when "field>=value"
-              value_at_path.to_f >= rule_value.to_f
-            when "field<value"
-              value_at_path.to_f < rule_value.to_f
-            when "field<=value"
-              value_at_path.to_f <= rule_value.to_f
-            when "field==value"
-              value_at_path.to_s == rule_value.to_s
-            when "field!=value"
-              value_at_path.to_s != rule_value.to_s
-            else
-              raise "Invalid type of #{rule['type']} in TriggerAgent##{id}"
+          if rule['type'] == 'not in'
+            rule_values.include?(value_at_path.to_s)
+          else
+            rule_values.any? do |rule_value|
```

Comment:
Could you convert this to an `if`...`elsif`` block instead of wrapping it in a new `if`?